File: kotlin/sprint5/N/Solution.kt
/**
 * Comment it before submitting
 *
 * class Node(var left: Node?, var right: Node?, var value: Int, var size: Int)
 */

fun split(root: Node?, k: Int): List<Node?> {
    // Your code
    // “ヽ(´▽｀)ノ”
}


fun test() {
    val node1 = Node(null, null, 3, 1)
    val node2 = Node(null, node1, 2, 2)
    val node3 = Node(null, null, 8, 1)
    val node4 = Node(null, null, 11, 1)
    val node5 = Node(node3, node4, 10, 3)
    val node6 = Node(node2, node5, 5, 6)
    val res: List<Node?> = split(node6, 4)
    assert(res[0]!!.size == 4)
    assert(res[1]!!.size == 2)
}

File: swift/sprint5/N/solution.swift
#if !REMOTE_JUDGE
open class Node {
    
    public var value: Int
    public var left: Node?
    public var right: Node?
    public var size: Int
    
    public init(_ value: Int) {
        self.value = value
        self.right = nil
        self.left = nil
        self.size = 1
    }
}
#endif

func calcSize(node: Node?) -> Int {
    if node == nil {
        return 0
    }
    return 1 + getSize(node: node!.left) + getSize(node: node!.right)
}

func getSize(node: Node?) -> Int {
    if node == nil {
        return 0
    }
    return node!.size
}

func split(root: Node?, k: Int) -> (Node?, Node?) {
    // Your code
    // ヽ(´▽`)/
    return (nil, nil)
}

func test() {
    var node1 = Node(3)
    var node2 = Node(2)
    node2.right = node1
    node2.size = 2
    var node3 = Node(8)
    
    var node4 = Node(11)
    var node5 = Node(10)
    node5.left = node3
    node5.right = node4
    var node6 = Node(5)
    node6.left = node2
    node6.right = node5
    var (left, right) = split(root: node6, k: 4)
    assert(left!.size == 4)
    assert(right!.size == 2)
}
